lib/mayaHydra/hydraExtensions/mixedUtils.h

```diff
@@ -30,8 +30,62 @@
 #include <maya/MMatrix.h>
 #include <maya/MObject.h>
 
+#if defined(_WIN32)
+#include <windows.h>
+#include "psapi.h"
+#elif defined(__linux__)
+#include <fstream>
+#include <sstream>
+#elif defined(__APPLE__)
+#include <mach/task.h>
+#include <mach/mach.h>
+#endif
+
 namespace MAYAHYDRA_NS_DEF {
 
+constexpr int MB = 1024 * 1024;
+
+// Function to get memory usage of the current process
+MAYAHYDRALIB_API
+inline int getProcessMemory()
+{
+#if defined(_WIN32)
+    // see https://learn.microsoft.com/en-us/windows/win32/api/psapi/nf-psapi-getprocessmemoryinfo
+    PROCESS_MEMORY_COUNTERS_EX pmc;
+    GetProcessMemoryInfo(GetCurrentProcess(), (PROCESS_MEMORY_COUNTERS*)&pmc, sizeof(pmc));
+    return static_cast<int>(pmc.WorkingSetSize / MB); // Working Set Size in MB
```

Comment:
Please get the Virtual memory (pmc.PrivateUsage) used by current process, instead of physical memory.